File: get_user_id.py
#!/usr/bin/env python3
"""
SlackユーザーID取得ヘルパースクリプト
「いち l AI就活（代表）」のユーザーIDを取得するためのスクリプト
"""

import os
from slack_sdk.web import WebClient

def get_user_id_by_name():
    """
    ユーザー名からユーザーIDを取得する
    """
    # 環境変数からトークンを取得
    slack_token = os.getenv('SLACK_API_TOKEN')
    
    if not slack_token:
        print("エラー: SLACK_API_TOKENが設定されていません")
        print("環境変数を設定するか、以下のように実行してください：")
        print("SLACK_API_TOKEN=your_token python get_user_id.py")
        return
    
    client = WebClient(token=slack_token)
    
    try:
        # 全ユーザー情報を取得
        response = client.users_list()
        users = response['members']
        
        print("=== ユーザー一覧 ===")
        print("検索キーワード: 'いち' を含むユーザー")
        print()
        
        found_users = []
        
        for user in users:
            real_name = user.get('real_name', '')
            display_name = user.get('display_name', '')
            user_id = user.get('id', '')
            
            # 「いち」を含むユーザーを検索
            if 'いち' in real_name or 'いち' in display_name:
                found_users.append({
                    'id': user_id,
                    'real_name': real_name,
                    'display_name': display_name
                })
                print(f"✅ 見つかりました:")
                print(f"   ユーザーID: {user_id}")
                print(f"   実名: {real_name}")
                print(f"   表示名: {display_name}")
                print()
        
        if not found_users:
            print("❌ 'いち' を含むユーザーが見つかりませんでした")
            print()
            print("全ユーザー一覧:")
            for user in users[:10]:  # 最初の10人を表示
                real_name = user.get('real_name', '')
                display_name = user.get('display_name', '')
                user_id = user.get('id', '')
                print(f"  {real_name} ({display_name}): {user_id}")
            
            if len(users) > 10:
                print(f"  ... 他 {len(users) - 10} 人のユーザー")
        
        print()
        print("=== GitHub Secrets設定用 ===")
        if found_users:
            print("以下の値をGitHub Secretsの 'SLACK_DM_USER_ID' に設定してください:")
            for user in found_users:
                print(f"  {user['id']}")
        else:
            print("ユーザーが見つからない場合は、手動でユーザーIDを確認してください")
            
    except Exception as e:
        print(f"エラーが発生しました: {e}")

if __name__ == "__main__":
    get_user_id_by_name()

File: get_user_id.js
import { WebClient } from '@slack/web-api';

// Env vars
const slackToken = process.env.SLACK_BOT_TOKEN;

if (!slackToken) {
  console.error('Missing SLACK_BOT_TOKEN');
  console.log('使用方法: SLACK_BOT_TOKEN=your_token node get_user_id.js');
  process.exit(1);
}

const client = new WebClient(slackToken);

/**
 * ユーザー名からユーザーIDを検索
 * @param {string} displayName - 表示名（例: "よしき", "りょうせい"）
 * @returns {string|null} ユーザーID
 */
async function findUserByName(displayName) {
  try {
    const result = await client.users.list();
    
    if (!result.ok) {
      console.error('❌ ユーザーリストの取得に失敗しました:', result.error);
      return null;
    }

    const users = result.members || [];
    
    // 表示名で検索
    const user = users.find(u => 
      u.profile?.display_name?.includes(displayName) ||
      u.profile?.real_name?.includes(displayName) ||
      u.name?.includes(displayName)
    );

    if (user) {
      return user.id;
    }

    return null;
  } catch (error) {
    console.error('❌ ユーザー検索中にエラーが発生しました:', error);
    return null;
  }
}

/**
 * 全ユーザーリストを表示
 */
async function listAllUsers() {
  try {
    const result = await client.users.list();
    
    if (!result.ok) {
      console.error('❌ ユーザーリストの取得に失敗しました:', result.error);
      return;
    }

    const users = result.members || [];
    
    console.log('📋 Slackワークスペースの全ユーザー:');
    console.log('=' .repeat(80));
    
    users.forEach(user => {
      if (!user.deleted && !user.is_bot) {
        const displayName = user.profile?.display_name || user.profile?.real_name || user.name || '名前なし';
        console.log(`👤 ${displayName}`);
        console.log(`   ID: ${user.id}`);
        console.log(`   メール: ${user.profile?.email || 'なし'}`);
        console.log('   ' + '-'.repeat(50));
      }
    });
    
  } catch (error) {
    console.error('❌ ユーザーリスト取得中にエラーが発生しました:', error);
  }
}

/**
 * 特定のユーザーを検索
 */
async function searchSpecificUsers() {
  const targetUsers = ['よしき', 'りょうせい'];
  
  console.log('🔍 対象ユーザーの検索結果:');
  console.log('=' .repeat(50));
  
  for (const userName of targetUsers) {
    const userId = await findUserByName(userName);
    if (userId) {
      console.log(`✅ ${userName}: ${userId}`);
    } else {
      console.log(`❌ ${userName}: 見つかりませんでした`);
    }
  }
}

/**
 * メイン処理
 */
async function main() {
  console.log('🚀 SlackユーザーID取得ツールを開始します...\n');
  
  // 特定ユーザーの検索
  await searchSpecificUsers();
  
  console.log('\n' + '=' .repeat(80));
  
  // 全ユーザーリストの表示
  await listAllUsers();
  
  console.log('\n📝 使用方法:');
  console.log('1. 上記の結果から対象ユーザーのIDをコピー');
  console.log('2. scripts/ryuukuru-x-reminder.js の MENTION_USERS を更新');
  console.log('3. 例: yoshiki: \'U0123456789\', ryosei: \'U0987654321\'');
}

// エラーハンドリング
main().catch((error) => {
  console.error('❌ 予期しないエラーが発生しました:', error);
  process.exit(1);
});
